server/routes/users.js

```diff
@@ -1,11 +1,88 @@
 const express = require('express');
 const router = express.Router();
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+const config = require('config');
+const {
+	body,
+	validationResult,
+} = require('express-validator');
+
+const User = require('../models/User');
 
 // @route   POST api/users
 // @desc    Register a user
 // @access  Public
-router.post('/', (req, res) => {
-	res.send('Register a user');
-});
+router.post(
+	'/',
+	[
+		body('firstname', 'Please include your first name')
+			.not()
+			.isEmpty(),
+		body('lastname', 'Please include your last name')
+			.not()
+			.isEmpty(),
+		body('email', 'Please include a valid email'),
+		body(
+			'password',
+			'Please include a password with 6 or more characters'
+		).isLength({ min: 6 }),
+	],
+	async (req, res) => {
+		const errors = validationResult(req);
+		if (!errors.isEmpty()) {
+			return res
+				.status(400)
+				.json({ errors: errors.array() });
+		}
+
+		const { firstname, lastname, email, password } =
+			req.body;
```

Comment:
No way to set a bio. Is this API intended to just to the bare minimum to make a user and later there will be a webpage to edit/set details on the user? If so then no change needed.